src/cdk/v2/destinations/pinterest_tag/procWorkflow.yaml

```diff
@@ -41,6 +43,13 @@ steps:
     condition: $.outputs.messageType === {{$.EventType.TRACK}}
     template: |
       $.assert(.message.event || .message.name, "event_name could not be mapped. Aborting.")
+
+  - name: isAdvertisingMappingEnabled
+    template: |
+      const featureFlagService = await $.createFeatureFlagService();
+      const pinterestAdFlag = await featureFlagService.isFeatureEnabled({ workspaceId: .metadata.workspaceId },'enable-pinterest-advertising-tracking-mapping');
```

Comment:
If possible, lets use constants for feature flag names - `enable-pinterest-advertising-tracking-mapping`